From khali@linux-fr.org Sun Dec 18 07:48:09 2005
Date: Sun, 18 Dec 2005 16:49:30 +0100
From: Jean Delvare <khali@linux-fr.org>
To: Greg KH <greg@kroah.com>
Cc: Laurent Riffard <laurent.riffard@free.fr>, Michael Geng <linux@MichaelGeng.de>, Petr Vandrovec <vandrove@vc.cvut.cz>
Subject: [PATCH 06/11] i2c: drop empty i2c_driver.command implementations
Message-Id: <20051218164930.658747df.khali@linux-fr.org>
Content-Disposition: inline; filename=i2c-drop-useless-command-callbacks.patch

From: Laurent Riffard <laurent.riffard@free.fr>

Given that implementing i2c_driver.command is optional, there is no
point in an empty implementation thereof.

Signed-off-by: Laurent Riffard <laurent.riffard@free.fr>
Signed-off-by: Jean Delvare <khali@linux-fr.org>
Cc: Michael Geng <linux@MichaelGeng.de>
Cc: Petr Vandrovec <vandrove@vc.cvut.cz>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/i2c/i2c-dev.c                 |    7 -------
 drivers/media/video/saa5246a.c        |    7 -------
 drivers/media/video/saa5249.c         |    7 -------
 drivers/video/matrox/matroxfb_maven.c |    5 -----
 4 files changed, 26 deletions(-)

--- gregkh-2.6.orig/drivers/i2c/i2c-dev.c
+++ gregkh-2.6/drivers/i2c/i2c-dev.c
@@ -464,12 +464,6 @@ static int i2cdev_detach_client(struct i
 	return 0;
 }
 
-static int i2cdev_command(struct i2c_client *client, unsigned int cmd,
-                           void *arg)
-{
-	return -1;
-}
-
 static struct i2c_driver i2cdev_driver = {
 	.driver = {
 		.name	= "dev_driver",
@@ -478,7 +472,6 @@ static struct i2c_driver i2cdev_driver =
 	.attach_adapter	= i2cdev_attach_adapter,
 	.detach_adapter	= i2cdev_detach_adapter,
 	.detach_client	= i2cdev_detach_client,
-	.command	= i2cdev_command,
 };
 
 static struct i2c_client i2cdev_client_template = {
--- gregkh-2.6.orig/drivers/media/video/saa5246a.c
+++ gregkh-2.6/drivers/media/video/saa5246a.c
@@ -151,12 +151,6 @@ static int saa5246a_detach(struct i2c_cl
 	return 0;
 }
 
-static int saa5246a_command(struct i2c_client *device, unsigned int cmd,
-	void *arg)
-{
-	return -EINVAL;
-}
-
 /*
  *	I2C interfaces
  */
@@ -169,7 +163,6 @@ static struct i2c_driver i2c_driver_vide
 	.id 		= I2C_DRIVERID_SAA5249, /* in i2c.h */
 	.attach_adapter = saa5246a_probe,
 	.detach_client  = saa5246a_detach,
-	.command 	= saa5246a_command
 };
 
 static struct i2c_client client_template = {
--- gregkh-2.6.orig/drivers/media/video/saa5249.c
+++ gregkh-2.6/drivers/media/video/saa5249.c
@@ -226,12 +226,6 @@ static int saa5249_detach(struct i2c_cli
 	return 0;
 }
 
-static int saa5249_command(struct i2c_client *device,
-			     unsigned int cmd, void *arg)
-{
-	return -EINVAL;
-}
-
 /* new I2C driver support */
 
 static struct i2c_driver i2c_driver_videotext = 
@@ -242,7 +236,6 @@ static struct i2c_driver i2c_driver_vide
 	.id 		= I2C_DRIVERID_SAA5249, /* in i2c.h */
 	.attach_adapter = saa5249_probe,
 	.detach_client  = saa5249_detach,
-	.command 	= saa5249_command
 };
 
 static struct i2c_client client_template = {
--- gregkh-2.6.orig/drivers/video/matrox/matroxfb_maven.c
+++ gregkh-2.6/drivers/video/matrox/matroxfb_maven.c
@@ -1288,10 +1288,6 @@ static int maven_detach_client(struct i2
 	return 0;
 }
 
-static int maven_command(struct i2c_client* client, unsigned int cmd, void* arg) {
-	return -ENOIOCTLCMD;	/* or -EINVAL, depends on who will call this */
-}
-
 static struct i2c_driver maven_driver={
 	.driver = {
 		.name	= "maven",
@@ -1299,7 +1295,6 @@ static struct i2c_driver maven_driver={
 	.id		= I2C_DRIVERID_MGATVO,
 	.attach_adapter	= maven_attach_adapter,
 	.detach_client	= maven_detach_client,
-	.command	= maven_command,
 };
 
 /* ************************** */
